### PR TITLE
Feature/add database test

### PR DESCRIPTION
**Pull request by ILoveMyTortoise**

### CODE REVIEW

File: src/Nox.Generator/Infrastructure/Persistence/DbContextGenerator/DbContextGenerator.cs [hunk 1]
```diff
@@ -83,8 +83,9 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 }
             }
             
-            base.OnModelCreating(modelBuilder);
         }
+
```
Comment (rochar) [inline src/Nox.Generator/Infrastructure/Persistence/DbContextGenerator/DbContextGenerator.cs:87]: this changes did not produced a new SampleWeAppDbContext?

**Review by alexandervlasenko — DISMISSED** [review]

**Review by rochar — APPROVED** [review]
approve it, seems good, just a few comments

File: src/Nox.Generator/Infrastructure/Persistence/DbContextGenerator/DbContextGenerator.cs (hunk 1 above)
Comment (ILoveMyTortoise) [reply src/Nox.Generator/Infrastructure/Persistence/DbContextGenerator/DbContextGenerator.cs:87]: Pushed generated code

File: tests/Nox.Generator.Tests/Database/EntityFrameworkTests.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
+﻿using FluentAssertions;
+using Nox.Types;
+using NoxSourceGeneratorTests.DatabaseTests;
+using SampleWebApp.Domain;
+using System.Linq;
+using Xunit;
+
+namespace Nox.Generator.Test.DatabaseTests;
+
+public class EntityFrameworkTests : SqliteTestBase
```
[the remark this answers is not in the document]
Comment (ILoveMyTortoise) [reply tests/Nox.Generator.Tests/Database/EntityFrameworkTests.cs]: renamed

File: tests/Nox.Generator.Tests/Database/SqliteTestBase.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+using System;
+
+namespace NoxSourceGeneratorTests.DatabaseTests;
+
+public abstract class SqliteTestBase : IDisposable
+{
+    private const string _inMemoryConnectionStringTemplate = "DataSource=:memory:";
+    //private const string _inMemoryConnectionStringTemplate = @"DataSource=test_database_{0}.db";
+    private static string _inMemoryConnectionString = string.Empty;
+    private const string _testSolutionFile = @"../../../Database/Design/test.solution.nox.yaml";
```
[the remark this answers is not in the document]
Comment (ILoveMyTortoise) [reply tests/Nox.Generator.Tests/Database/SqliteTestBase.cs]: change relative path

File: tests/Nox.Generator.Tests/Database/Design/test.solution.nox.yaml [hunk 4]
```diff
@@ -0,0 +1,237 @@
+﻿#
+# workplace.solution.nox.yaml
+#
+# yaml-language-server: $schema=https://noxorg.dev/schemas/solution.json
+#
+
+name: SampleWebApp
```
[the remark this answers is not in the document]
Comment (ILoveMyTortoise) [reply tests/Nox.Generator.Tests/Database/Design/test.solution.nox.yaml]: renamed

**Review by alexandervlasenko — APPROVED** [review]

**Pull request merged**